### PR TITLE
Add a function to create ProcessGroupUCC in PyTorch

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This PR will be used in https://github.com/pytorch/pytorch/pull/69564. This PR can be landed separately and doesn't need to wait for https://github.com/pytorch/pytorch/pull/69564.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
Ready for review

File: src/torch_ucc_init_oss.cpp [hunk 1]
```diff
@@ -1,4 +1,19 @@
 #include <torch/python.h>
+#include "torch_ucc.hpp"
 
-PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
+// This function is intentionally designed to take a void * argument
```
Comment (dzhulgakov) [inline src/torch_ucc_init_oss.cpp:4]: can you just use `export "C"`? it'd disable mangling

Comment (zasdfgbnm) [reply src/torch_ucc_init_oss.cpp]: It works! Thanks! I didn't know this trick.

**Comment by zasdfgbnm** [comment]
cc: @pallab-zz @kingchc @srinivas212 for review. This will be needed for https://github.com/pytorch/pytorch/pull/69564

File: src/torch_ucc_init_oss.cpp [hunk 2]
```diff
@@ -1,4 +1,13 @@
 #include <torch/python.h>
+#include "torch_ucc.hpp"
 
-PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
+using namespace c10d;
+
+extern "C" C10_EXPORT c10::intrusive_ptr<ProcessGroup> createProcessGroupUCCForNCCL(
```
Comment (kingchc) [inline src/torch_ucc_init_oss.cpp:6]: minor - might be good to use a more generic function name 
```suggestion
extern "C" C10_EXPORT c10::intrusive_ptr<ProcessGroup> createProcessGroupUCC(
```

**Review by kingchc — APPROVED** [review]

File: src/torch_ucc_init_oss.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply src/torch_ucc_init_oss.cpp]: Changed

**Pull request closed without merging**